substrate/frame/treasury/src/lib.rs

```diff
@@ -296,6 +305,9 @@ pub mod pallet {
 	pub type Deactivated<T: Config<I>, I: 'static = ()> =
 		StorageValue<_, BalanceOf<T, I>, ValueQuery>;
 
+	/// DEPRECATED: associated with `spend_local` call and will be removed in May 2025.
```

Comment:
damn, seems like you cannot use https://github.com/paritytech/polkadot-sdk/pull/4948

Reply:
Yup, it still shows warning, I tried to make it work in every order possible with `#[deprecated]` annotation

Reply:
@pkhry @jsdw wdyt? I think the consts in metadata should also have deprecation, and it would be quite useful to have them especially as we might gradually migrate a lot of consts in metadata to view functions. 

Was there a specific reason that they ended up not supporting deprecation?

Reply:
Deprecation attribute should be parsed for items annotated with [pallet::constant] ill take a look tomorrow

Reply:
Okay, then we might not merge this until this is done, it will be very good to properly add a deprecation metadata to the const metadata as well.

Reply:
The warning from understand comes forms he fact that we still propagate the deprecation attribute to the expanded code. I'll submit a pr tomorrow that will remove those attributes from expanded code

Reply:
Waiting on https://github.com/paritytech/polkadot-sdk/pull/6312 🙏

Reply:
> Deprecation attribute should be parsed for items annotated with [pallet::constant] ill take a look tomorrow

What about this? will this stuff properly go into metadata as deprecated?

What we want here is ideally PJS API to show `api.const.max_approvals` as deprecated too.

Reply:
We won't wait for https://github.com/paritytech/polkadot-sdk/pull/6312 to get into next polkadot release

Reply:
> > Deprecation attribute should be parsed for items annotated with [pallet::constant] ill take a look tomorrow
> 
> What about this? will this stuff properly go into metadata as deprecated?
> 
> What we want here is ideally PJS API to show `api.const.max_approvals` as deprecated too.

Yup, items annotated by `storage` will also collect deprecation info inside metadata

Reply:
My question is about `#[pallet::constant]`, not `#[pallet::storage]` :)

Reply:
> My question is about `#[pallet::constant]`, not `#[pallet::storage]` :)

I'm now officially blind I guess, yeah.it will pick up the constants and deprecation info from them